### ISSUE TITLE
Learning rate scheduling with Flax does not work

### ISSUE THREAD

**Issue by dirkgr** [allenai/tango#477]
### 🚀 The feature, motivation and pitch

The torch trainer has LR scheduling. The flax trainer should as well.

### Alternatives

* None

### Additional context

The trainer takes an `lr_scheduler` argument, but ignores it.

**Comment by SauravMaheshkar** [comment]
The `FlaxTrainStep` does have a `lr_scheduler` argument. I can't understand the issue can you provide more context ??

https://github.com/allenai/tango/blob/1ee2c56a0f557de3b2ba0bb25432487f0c4c50b5/tango/integrations/flax/train.py#L84

**Comment by dirkgr** [comment]
If you chase down where that is actually used, you find this line: https://github.com/allenai/tango/blob/1ee2c56a0f557de3b2ba0bb25432487f0c4c50b5/tango/integrations/flax/train.py#L294
